cassdegrees/ui/views.py

```diff
@@ -224,73 +223,100 @@ def manage_courses(request):
         'msg': None,
         'is_error': False
     }
+
     if request.method == 'POST':
         model_api_url = request.build_absolute_uri('/api/model/course/')
         post_data = request.POST
-        # actual_request = post_data.get('_method')
-
-        if action == 'Add':
-            # courses = [{'code': course} for course in set([x['code'] for x in courses])]
-            # Create a python dictionary with exactly the same fields as the model (in this case, CourseModel)
-            offered_sems = post_data.getlist('semesters[]')
-            course_instance = \
-                {
-                    'code': post_data.get('code'),
-                    'year': post_data.get('year'),
-                    'name': post_data.get('name'),
-                    'units': post_data.get('units'),
-                    'offeredSem1': 'semester1' in offered_sems,
-                    'offeredSem2': 'semester2' in offered_sems
-                }
-            # Submit a POST request to the course API with course_instance as data
-            rest_api = requests.post(model_api_url, data=course_instance)
-            if rest_api.status_code == 201:
-                render_properties['msg'] = 'Course successfully added!'
-            else:
-                render_properties['is_error'] = True
-                # detects if the course already exists
-                if 'The fields code, year must make a unique set.' in rest_api.json()['non_field_errors']:
-                    render_properties['msg'] = "The course you are trying to create already exists!"
+        request_origin = post_data.get('request_origin')
+        print(request_origin)
+
+        if request_origin == 'managecourses.html':
```

Comment:
I think you need to add some documentation here about what the specific differences are for this. It wasn't immediately clear to me what this suggested.

Reply:
Added some comments explaining this.